### PR TITLE
Add proto matcher

### PR DESCRIPTION
**Pull request by niklasembretsen**

### CODE REVIEW

File: gomockmatchersproto/proto_test_message.proto [hunk 1]
```diff
@@ -0,0 +1,10 @@
+syntax = "proto3";
+
+option go_package = "./gomockmatchers";
+
+package gomockmatchers;
+
+message TestRecord {
```
Comment (niklasembretsen) [inline gomockmatchersproto/proto_test_message.proto:7]: Not sure if there is any other way to make this testable than adding an actual proto message here?

File: proto.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+func (p *ProtoMatcher) Matches(x interface{}) bool {
+	pbx, ok := x.(proto.Message)
+	if !ok {
+		return false
+	}
+	return proto.Equal(p.data, pbx)
+}
+
+func (p *ProtoMatcher) String() string {
+	return "is matching"
```
Comment (niklasembretsen) [inline proto.go:25]: This is the most sophisticated `String()` function I could come up with... not much useful info in `proto.Message`

**Comment by niklasembretsen** [comment]
I wonder if the pending workflow is something old that is configured on this repo? I cant find anything... 🤔

File: proto.go (hunk 2 above)
Comment (noseglid) [reply proto.go:25]: This string is supposed to explain what is in `p.data` - perhaps a JSON serialization? Something like

```go
data, err := json.Marshal(p.data)
if err != nil {
  return "<unknown>"
}
return data
```

File: gomockmatchersproto/proto_test_message.proto (hunk 1 above)
Comment (noseglid) [reply gomockmatchersproto/proto_test_message.proto:7]: This is a bit annoying and cluttering.

Can we create a mock of the `proto.Message` using gomock instead?

Comment (niklasembretsen) [reply gomockmatchersproto/proto_test_message.proto]: I made use of the [structpb](https://pkg.go.dev/google.golang.org/protobuf/types/known/structpb) instead! (wasnt really sure how to go about mocking `proto.Message` tbh)

**Review by noseglid — APPROVED** [review]
clean! ⭐

**Pull request merged**